pkgs/development/python-modules/laszip/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, scikit-build-core
+, distlib
+, pytestCheckHook
+, pyproject-metadata
+, pathspec
+, pybind11
+, cmake
+, LASzip
+}:
+
+buildPythonPackage rec {
+  pname = "laszip-python";
+  version = "0.2.1";
+
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    rev = version;
+    repo = pname;
+    owner = "tmontaigu";
+    sha256 = "sha256-ujKoUm2Btu25T7ZrSGqjRc3NR1qqsQU8OwHQDSx8grY=";
+  };
+
+  propagatedBuildInputs = [
+    scikit-build-core
+    distlib
+    pyproject-metadata
+    pathspec
```

Comment:
I believe some of these are actually build-time only dependencies (scikit-build-core is one of them), and they should be moved to nativeBuildInputs.

Reply:
And the later three are just identical to `scikit-build-core.optional-dependencies.pyproject`

Reply:
The proper usage can be like in `pkgs/development/python-modules/awkward-cpp/default.nix`

Reply:
Thank you, good catch @NickCao